apps/glusterfs/volume_entry.go

```diff
@@ -418,11 +418,34 @@ func (v *VolumeEntry) cleanupCreateVolume(db wdb.DB,
 
 	// from a quick read its "safe" to unconditionally try to delete
 	// bricks. TODO: find out if that is true with functional tests
-	DestroyBricks(db, executor, brick_entries)
+	reclaimed, err := DestroyBricks(db, executor, brick_entries)
+	if err != nil {
+		logger.LogError("failed to destory bricks: %v", err)
+	}
 	return db.Update(func(tx *bolt.Tx) error {
 		for _, brick := range brick_entries {
 			v.removeBrickFromDb(tx, brick)
 		}
+		// update the device' free/used space after removing bricks
+		for _, b := range brick_entries {
+			if reclaim, ok := reclaimed[b.Info.DeviceId]; ok {
```

Comment:
Minor nit: I like using a variable name like "found" vs "ok" for cases like this as a reminder that the 2nd return value tests the presence of the key in the map.